### PR TITLE
Browser-based previews with express server.

### LINKED ISSUE #25: [Suggestion] - Easy browser-based preview UI.
Thinking a quick and easy way to do browser based testing would be a sweet addition to this project.
I've done the proof of concept locally and feel like its a good addition to this project.

__Why?__ 
* Don't have to send an email for every change
* No need to remember local url's to navigate to (would load from `localhost:[PORT]`)
* Side-by side desktop / mobile preview
* Lightweight

__Implementation Thoughts__
* Implement express server w/ grunt
* Live reload for changes to /dist files
* Grunt open to launch a browser (optional)
* Serve dist files into the ui example below for quick side-by-side previews

__Here's a screenshot of what I cooked up locally as proof of concept:__
screen shot 2015-03-21 at 12 04 06 pm

### PR DESCRIPTION
**Pull request by taeo**
Ready to go. Fixes #25.

### CODE REVIEW

**Comment by leemunroe** [comment]
@taeo Ping to see if you want to bring this up to date with master. Think it will be a great addition.

**Comment by taeo** [comment]
@leemunroe - Yes, sorry for the silence. I'll get this up to date with master and send the PR.

Cheers

**Comment by chrisrowe** [comment]
@taeo @leemunroe Thanks so much for this, great work. Looking forward to the PR merging :pray:

**Comment by taeo** [comment]
'aight @leemunroe - This looks good on my end and is up to date. 
FYI @chrisrowe

File: Gruntfile.js [hunk 1]
```diff
@@ -272,4 +339,13 @@ module.exports = function(grunt) {
...
     grunt.registerTask('s3upload', ['aws_s3:prod', 'cdn:aws_s3']);
 
+    // Launch the express server and start watching
+    // NOTE: The server will not stay running if the grunt watch task is not active
+    grunt.registerTask('serve', ['default', 'express', 'open', 'watch']);
+
+
+    // Launch the express server and start watching
+    // NOTE: The server will not stay running if the grunt watch task is not active
+    grunt.registerTask('serve', ['default', 'autoprefixer:preview', 'express', 'open', 'watch']);
```
Comment (leemunroe) [inline Gruntfile.js]: Can we remove line 344? I think these are duplicates if I'm not mistaken.

**Comment by leemunroe** [comment]
Looking awesome :+1:

**Comment by taeo** [comment]
@leemunroe - Good catch. I caught that on my master, but never hit the feature branch. just cherry picked it and should be good to go!

File: Gruntfile.js [hunk 2]
```diff
@@ -261,7 +328,7 @@ module.exports = function(grunt) {
     require('load-grunt-tasks')(grunt);
 
     // Where we tell Grunt what to do when we type "grunt" into the terminal.
-    grunt.registerTask('default', ['sass','assemble','premailer','imagemin','replace:src_images']);
+    grunt.registerTask('default', ['sass','autoprefixer:preview','assemble','premailer','imagemin','replace:src_images']);
```
Comment (leemunroe) [inline Gruntfile.js]: Can we remove autoprefixer here?

Comment (taeo) [inline Gruntfile.js]: Only if you want less browser coverage. Not recommended.
Unless there's actual bugs / issues, i'd recommend going through with the PR and we can enhance later as needed.

Comment (leemunroe) [inline Gruntfile.js]: Autoprefixer is only used for the preview server though (grunt serve) right?

Comment (taeo) [inline Gruntfile.js]: Correct.

**Comment by leemunroe** [comment]
@taeo Great work 🙌

**Comment by chrisrowe** [comment]
😍 Thanks for all your hard work on this @leemunroe and @taeo. You've both saved me countless hours and hair pulling

**Comment by taeo** [comment]
Thank You! :bow:

**Pull request merged**